### PR TITLE
New Dagstermill Marshalling

### PR DESCRIPTION
**Pull request by schrockn**

### CODE REVIEW

**Comment by schrockn** [comment]
Looks like there is some error from switching from pickle to dill..

@mgasner have you ever seen this?

orig_message="Can't pickle <class 'botocore.client.S3'>: it's not found as botocore.client.S3

File: python_modules/dagstermill/dagstermill/dagstermill_tests/test_dagstermill_manager.py [hunk 1]
```diff
@@ -13,125 +13,18 @@
     types,
 )
 
+# def test_basic_get_in_memory_input():
```
Comment (mgasner) [inline python_modules/dagstermill/dagstermill/dagstermill_tests/test_dagstermill_manager.py]: why not retire these completely?

File: python_modules/dagma/tests/test_lambda_engine.py [hunk 2]
```diff
@@ -138,13 +138,15 @@ def run_test_pipeline(pipeline):
             return results
 
 
+@pytest.mark.skip('does not seem to work')
```
Comment (mgasner) [inline python_modules/dagma/tests/test_lambda_engine.py]: wait, what's happening here?

File: python_modules/airline-demo/airline_demo/notebooks/Delays by Geography.ipynb [hunk 3]
```diff
@@ -11,8 +11,11 @@
    "outputs": [],
    "source": [
     "import dagstermill as dm\n",
-    "manager = dm.define_manager()\n",
-    "inputs = manager.define_inputs(db_url='', eastbound_delays='', westbound_delays='')\n",
+    "from airline_demo.pipelines import define_repo\n",
```
Comment (mgasner) [inline python_modules/airline-demo/airline_demo/notebooks/Delays by Geography.ipynb]: this is much nicer

**Comment by mgasner** [comment]
I believe the dagma issue you're seeing is in pickling/unpickling the new TransformExecutionInfo, which now has a logger buried inside of it https://github.com/dagster-io/dagster/pull/486/files#diff-e7a1c6dc11dd440695a29a83aa3303d2R1179. Since making the pipeline defs serializable is equivalent to just implementing the pickle protocol (__getstate__), I think we should maybe bite the bullet on that

**Comment by mgasner** [comment]
Resolve merge conflicts then go ahead and merge while we figure out a new strategy for dagma.

**Pull request merged**